commit 05b0c46baa0f5ab9feca1ad696eefd2a66425b04 (refs/changes/46/3546/3)
Author: John Levon <john.levon@joyent.com>
Date:   2018-03-06T12:55:44+00:00 (1 year, 7 months ago)
    
    OS-6728 bhyve /dev should be cleaned out
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Approved by: Jerry Jelinek <jerry.jelinek@joyent.com>

diff --git a/usr/src/lib/brand/bhyve/zone/platform.xml b/usr/src/lib/brand/bhyve/zone/platform.xml
index 681960c73c..da7740c4f6 100644
--- a/usr/src/lib/brand/bhyve/zone/platform.xml
+++ b/usr/src/lib/brand/bhyve/zone/platform.xml
@@ -37,53 +37,13 @@
 	    opt="size=10m" />
 
 	<!-- Devices to create under /dev -->
-	<device match="arp" />
-	<device match="bpf" />
-	<device match="conslog" />
-	<device match="cpu/self/cpuid" />
-	<device match="crypto" />
-	<device match="cryptoadm" />
-	<device match="dsk" />
-	<device match="dtrace/*" />
-	<device match="dtrace/provider/*" />
+	<device match="dld" ip-type="exclusive" />
 	<device match="fd" />
-	<device match="ipnet" />
-	<device match="kstat" />
-	<device match="lo0" />
-	<device match="lofictl" />
-	<device match="lofi" />
-	<device match="log" />
-	<device match="logindmux" />
-	<device match="nsmb" />
-	<device match="net/*" />
 	<device match="null" />
-	<device match="openprom" arch="sparc" />
-	<device match="poll" />
-	<device match="pool" />
-	<device match="ptmx" />
-	<device match="pts/*" />
 	<device match="random" />
 	<device match="rdsk" />
-	<device match="rlofi" />
-	<device match="rmt" />
-	<device match="sad/user" />
-	<device match="svvslo0" />
-	<device match="svvslo1" />
-	<device match="svvslo2" />
-	<device match="svvslo3" />
-	<device match="swap" />
-	<device match="sysevent" />
-	<device match="tcp" />
-	<device match="tcp6" />
-	<device match="term" />
-	<device match="ticlts" />
-	<device match="ticots" />
-	<device match="ticotsord" />
-	<device match="tty" />
-	<device match="udp" />
-	<device match="udp6" />
-	<device match="urandom" />
 	<device match="viona" />
+
 	<!--
 	    Careful: this is not just about catching both /dev/vmmctl and the
 	    /dev/vmm directory.  In fact it's required to prevent a race where
@@ -92,51 +52,14 @@
 	    directory.  Adding the glob here ensures it's checked as needed.
 	-->
 	<device match="vmm*" />
-	<device match="vnd/*" />
-	<device match="zero" />
-	<device match="zfs" />
-
-	<!-- Devices to create in exclusive IP zone only -->
-	<device match="dld" ip-type="exclusive" />
-	<device match="icmp" ip-type="exclusive" />
-	<device match="icmp6" ip-type="exclusive" />
-	<device match="ip" ip-type="exclusive" />
-	<device match="ip6" ip-type="exclusive" />
-	<device match="ipauth" ip-type="exclusive" />
-	<device match="ipf" ip-type="exclusive" />
-	<device match="ipl" ip-type="exclusive" />
-	<device match="iplookup" ip-type="exclusive" />
-	<device match="ipmpstub" ip-type="exclusive" />
-	<device match="ipnat" ip-type="exclusive" />
-	<device match="ipscan" ip-type="exclusive" />
-	<device match="ipsecah" ip-type="exclusive" />
-	<device match="ipsecesp" ip-type="exclusive" />
-	<device match="ipstate" ip-type="exclusive" />
-	<device match="ipsync" ip-type="exclusive" />
-	<device match="keysock" ip-type="exclusive" />
-	<device match="rawip" ip-type="exclusive" />
-	<device match="rawip6" ip-type="exclusive" />
-	<device match="rts" ip-type="exclusive" />
-	<device match="sad/admin" ip-type="exclusive" />
-	<device match="sctp" ip-type="exclusive" />
-	<device match="sctp6" ip-type="exclusive" />
-	<device match="spdsock" ip-type="exclusive" />
-	<device match="sppp" ip-type="exclusive" />
-	<device match="sppptun" ip-type="exclusive" />
-	<device match="vni" ip-type="exclusive" />
 
 	<!-- Renamed devices to create under /dev -->
 	<device match="zcons/%z/zoneconsole" name="zconsole" />
 
 	<!-- Symlinks to create under /dev -->
 	<symlink source="console" target="zconsole" />
-	<symlink source="dtremote" target="/dev/null" />
-	<symlink source="msglog" target="zconsole" />
 	<symlink source="stderr" target="./fd/2" />
 	<symlink source="stdin" target="./fd/0" />
 	<symlink source="stdout" target="./fd/1" />
-	<symlink source="syscon" target="zconsole" />
-	<symlink source="sysmsg" target="zconsole" />
-	<symlink source="systty" target="zconsole" />
 
 </platform>
